src/Microsoft.DotNet.Interactive/InstallPackagesMessage.cs

```diff
@@ -0,0 +1,105 @@
+                int progress)
+        {
+            RestoreSources = restoreSources;
+            InstalledPackages = installedPackages;
+            InstallingPackages = installingPackages;
+            Progress = progress;
+        }
+
+        IHtmlContent InstallMessage(string message, IEnumerable<string> items, string progress = null)
+        {
+            if (items.Count() > 0)
+            {
+                return div(
+                    strong(message),
+                    ul(items.Select(s => li(span(s + progress)))));
+            }
+            else
+            {
+                return div();
+            }
+        }
+
+        private string FormatAsHtml()
+        {
+            string progress = new String('.', Progress);
+            var items = new List<IHtmlContent>();
+            items.Add(InstallMessage("Restore sources", RestoreSources));
+            items.Add(InstallMessage("Installed Packages", InstalledPackages));
+            items.Add(InstallMessage("Installing Packages", InstallingPackages, progress));
+            return div(items).ToDisplayString();
+        }
+
+        private string FormatAsPlainText()
+        {
+            var result = new StringBuilder();
+            if (RestoreSources.Count > 0)
+            {
+                result.Append("Restore sources");
+                foreach (var source in RestoreSources)
+                {
+                    result.Append(System.Environment.NewLine);
+                    result.Append(" - " + source);
+                }
+                result.Append(System.Environment.NewLine);
+            }
+
+            if (InstalledPackages.Count > 0)
+            {
+                result.Append("Installed Packages");
+                foreach (var installed in InstalledPackages)
+                {
+                    result.Append(System.Environment.NewLine);
+                    result.Append(" - " + installed);
+                }
+                result.Append(System.Environment.NewLine);
+            }
+
+            if (InstallingPackages.Count > 0)
+            {
+                result.Append("Installing Packages");
+                foreach (var installing in InstallingPackages)
+                {
+                    result.Append(System.Environment.NewLine);
+                    result.Append(" - " + installing + "   ");
+                    result.Append('.', Progress);
+                }
+                result.Append(System.Environment.NewLine);
+            }
+            return result.ToString();
+        }
+
+        static public void RegisterInstallPackagesMimeType(string mimeType)
+        {
+            Formatter.Register<InstallPackagesMessage>(m => m.FormatAsPlainText(), PlainTextFormatter.MimeType);
+            Formatter.Register<InstallPackagesMessage>(m => m.FormatAsHtml(), HtmlFormatter.MimeType);
+            Formatter.SetPreferredMimeTypeFor(typeof(InstallPackagesMessage), PlainTextFormatter.MimeType);
```

Comment:
The MIME type preference shouldn't be needed. This isn't the standard default and I would think HTML would be preferred.

Reply:
Actually, I like the plaintext better, it redraws without flashing.